enhancements/machine-api/aws-placement-groups.md

```diff
@@ -0,0 +1,487 @@
+  // This value is only used in conjunction with a `Partition` placement Group.
+  // It is recommended to only use this value if multiple MachineSets share
+  // a single Placement Group, in which case, each MachineSet should represent an individual partition number.
+  // If unset, when a Partition placement group is used, AWS will attempt to
+  // distribute instances evenly between partitions.
+  // +kubebuilder:validation:Minimum:=1
+  // +kubebuilder:validation:Maximum:=7
+  // +optional
+  PartitionNumber int32 `json:"partitionNumber,omitempty"`
+}
+
+// AWSPlacementGroupType represents the valid values for the Placement GroupType field.
+type AWSPlacementGroupType string
+
+const (
+  // AWSClusterPlacementGroupType is the "Cluster" placement group type.
+  // Cluster placement groups place instances close together to improve network latency and throughput.
+  AWSClusterPlacementGroupType   AWSPlacementGroupType = "Cluster"
+  // AWSPartitionPlacementGroupType is the "Partition" placement group type.
+  // Partition placement groups reduce the likelihood of hardware failures
+  // disrupting your application's availability.
+  // Partition placement groups are recommended for use with large scale
+  // distributed and replicated workloads.
+  AWSPartitionPlacementGroupType AWSPlacementGroupType = "Partition"
+  // AWSSpreadPlacementGroupType is the "Spread" placement group type.
+  // Spread placement groups place instances on distinct racks within the availability
+  // zone. This ensures instances each have their own networking and power source
+  // for maximum hardware fault tolerance.
+  // Spread placement groups are recommended for a small number of critical instances
+  // which must be kept separate from one another.
+  // Using a Spread placement group imposes a limit of seven instances within
+  // the placement group within a single availability zone.
+  AWSSpreadPlacementGroupType    AWSPlacementGroupType = "Spread"
+)
+```
+
+### Implementation Details/Notes/Constraints
+
+#### Placement Group Strategies
+
+AWS offers three different placement group strategies.
+Each strategy is designed with a different use case in mind and each has it's own caveats.
+The details of each are fleshed out below.
+
+##### Cluster Placement Groups
+
+A Cluster placement group is a grouping of instances within a single Availability zone.
+Cluster placement groups are used for workloads that require higher per-flow throughput limits for TCP/IP traffic and
+as such, are placed in the same high-bisection bandwidth segment of the network.
+
+Cluster placement groups are recommended for applications that benefit from low network latency, high network
+throughput, or both.
+In particular, we expect that these will be used in conjunction with EFA network interfaces in HPC applications.
+
+When using Cluster placement groups, AWS recommends to launch all instances as the same instance type and if
```

Comment:
will this be the strategy used by whatever fulfills our machine API?  if not, can you describe how we do it?

Reply:
This is expanded upon in the risks/mitigations, will add a note and a link here (#adding-new-instances-to-placement-groups-may-cause-capacity-errors)